### ISSUE TITLE
[bug] Explicitly listed path that starts with a dot is ignored.

### ISSUE THREAD

**Issue by ioquatix** [actions/upload-artifact#618]
### What happened?

Explicitly referenced files, e.g. `.coverage.db` are not uploaded.

### What did you expect to happen?

The file should be uploaded as it exists and is explicitly listed: <https://github.com/socketry/protocol-http/commit/036101536b693f1569f9b139934d4946cda7120d>

### How can we reproduce it?

Explicitly listing a path that starts with a dot.

### What version of the action are you using?

v4

### What are your runner environments?

linux

**Comment by ioquatix** [comment]
I understand the value of ignoring hidden files when using glob patterns, but ignoring a file which is explicitly listed with an error "Warning: No files were found with the provided path: .covered.db. No artifacts will be uploaded." is extremely confusing, and made me think something was wrong with my own software. Suggest the following:

- Indicate in the error that files were matched but ignored.
- If an explicit path is given and matches, it should not be ignored.

**Comment by ioquatix** [comment]
It might also be a nice idea to document somewhere what hidden files are, e.g. are we talking about dot files, `.gitignore`ed files, or anything else?

**Comment by abravalheri** [comment]
I think I am also hitting this problem in https://github.com/abravalheri/validate-pyproject/actions/runs/11034259945/job/30647361576.

![image](https://github.com/user-attachments/assets/18c1b84e-0e56-4062-9e8f-5bc254630fa2)


In this GHA step, I am making sure all files exist and that the directory to be uploaded is not empty:

https://github.com/abravalheri/validate-pyproject/compare/v0.20...v0.20.1

but even when they exist the `upload-artifact` action seems to be ignoring them.

**Comment by markfickett** [comment]
It would be great to log "Ignored hidden files `.env`, `.bin/`, and 9 others, use `include-hidden-files: true` if you want to include them.".

**Comment by defanator** [comment]
> It would be great to log "Ignored hidden files `.env`, `.bin/`, and 9 others, use `include-hidden-files: true` if you want to include them.".

Agree. Just stumbled upon this on one of self-hosted runners with updated action, all the logs were quite misleading (and enabling debug didn't help at all).

**Comment by daveisfera** [comment]
Yes, a warning out or a[llowing specific paths in hidden directories](https://github.com/daveisfera/test_upload-artifact/actions/runs/13931636803/job/38989949471), would be nice
